libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI+Blocks.m

```diff
@@ -231,23 +231,27 @@ - (SFRestRequest *) performRequestWithMethod:(SFRestMethod)method path:(NSString
 #pragma mark - response delegate
 
 - (void) sendActionForRequest:(SFRestRequest *)request success:(BOOL)success withObject:(id)object {
-    if( success ) {
-        // This block def basically generalizes the SFRestDictionaryResponseBlock and SFRestArrayResponseBlock
-        // block typedefs, so that we can handle either.
+
+    // Ensures that the response is a well-known type.
+    if (![object isKindOfClass:[NSDictionary class]] && ![object isKindOfClass:[NSArray class]] && ![object isKindOfClass:[NSString class]]) {
```

Comment:
But &lt;&gt; is an `NSString` too?  This feels a bit like it's masking whatever the root cause of the issue is.  Why can the consumer not handle output that's otherwise considered valid (if not exactly usual) from the service?